includes/class-wc-install.php

```diff
@@ -156,7 +156,7 @@ class WC_Install {
 	public static function init() {
 		add_action( 'init', array( __CLASS__, 'check_version' ), 5 );
 		add_action( 'init', array( __CLASS__, 'manual_database_update' ), 20 );
-		add_action( 'plugins_loaded', array( __CLASS__, 'wc_admin_db_update_notice' ), 100 );
+		add_action( 'woocommerce_admin_updated', array( __CLASS__, 'wc_admin_db_update_notice' ) );
```

Comment:
this is a very significant change because we are now hooked on an action that will happen only once in a lifetime per version from an action that was happening on every request. 

Do we also want to remove early return conditions from https://github.com/woocommerce/woocommerce/blob/master/includes/admin/notes/class-wc-notes-run-db-update.php#L29 to make ensure that `show_reminder` runs (or maybe call it directly from the constructor)?

Maybe another way would be to keep this as it is and add a check for if option `woocommerce_admin_install_timestamp` is present?

Reply:
> this is a very significant change because we are now hooked on an action that will happen only once in a lifetime per version from an action that was happening on every request.

Good point. 🤔 

> Maybe another way would be to keep this as it is and add a check for if option `woocommerce_admin_install_timestamp` is present?

Yeah I think this is a good approach - I'll work on modifying the PR.